### PR TITLE
Formatter: use alejandra

### PR DESCRIPTION
**Pull request by DavHau**
How about introducing alejandra as a formatter? @steveeJ Let me know what you think.

This [adds alejandra as a formatter via flake.nix](https://github.com/holochain/holochain-infra/commit/3ee73d0bec1e6bbb3837a38982262135216658f4) and formats all files

### CODE REVIEW

File: modules/apps.deploy-/darwin.nix [hunk 1]
```diff
@@ -1,41 +1,42 @@
 # populates apps.{system}.deploy-... for all darwin hosts
-{ self, lib, ... }: {
-  perSystem = { pkgs, ... }:
-    let
-      mkDarwinDeploy =
-        { attrName
-        , hostName
-        ,
-        }: pkgs.writeScript "deploy-${hostName}" ''
```
Comment (DavHau) [inline modules/apps.deploy-/darwin.nix:9]: nixpkgs-fmt really does weird things here for example

File: modules/_template.nix [hunk 2]
```diff
@@ -1,4 +1,14 @@
-{ self, lib, ... }: {
-  perSystem = { config, self', inputs', pkgs, ... }: { };
-  flake = { };
+{
+  self,
+  lib,
+  ...
+}: {
+  perSystem = {
+    config,
```
Comment (steveej) [inline modules/_template.nix:7]: can alejandra be configured to keep the commas at the beginning of the line?

Comment (DavHau) [reply modules/_template.nix:7]: Being configurable would go against the purpose of a formatter, which is to have the same format everywhere in the ecosystem.
Alejandra does not use leading commas. More details about that decision can be found here: https://github.com/kamadorueda/alejandra/issues/95

Comment (steveej) [reply modules/_template.nix:7]: what happens if you add a trailing comma, does it remove it?

Comment (DavHau) [reply modules/_template.nix:7]: Trailing commas are fine for alejandra. I have the habit to put them everywhere for consistency.

Comment (steveej) [reply modules/_template.nix:7]: yes i was wondering if it adds them when they're missing! i guess we'll find out

**Pull request merged**